src/binary/decimal.rs

```diff
@@ -111,12 +111,9 @@ where
         // We need to know its encoded length before we can write out
         // the preceding type descriptor.
         let mut encoded: ArrayVec<u8, DECIMAL_BUFFER_SIZE> = ArrayVec::new();
-        encoded
-            .encode_decimal(decimal)
-            .map_err(|_e| IonError::EncodingError {
-                description: "found a decimal that was too large for the configured buffer"
-                    .to_string(),
-            })?;
+        encoded.encode_decimal(decimal).map_err(|_e| {
+            IonError::encoding_error("found a decimal that was too large for the configured buffer")
```

Comment:
🗺️ Previously, the convenience methods for constructing errors would take either:
1. **`AsRef<str>`**, which accepts both `String` and `&str`, but required the method to make a copy of whatever was passed in so it could be stored indefinitely. If the user passed a `String`, we'd be making an unnecessary copy.
2. **`String`**, giving the user full control over how/when the `String` was constructed, and requiring at most one allocation. However, if the user didn't already have a `String`, they'd have to construct one before passing it in.

These methods now accept an `Into<String>`, allowing them to accept either a `String` or a `&str` and only making a new `String` when it's necessary.